pycs/gen/lc.py

```diff
@@ -2176,20 +2212,30 @@ def display(lclist=[], splist=[],
 				npts = (spline.datapoints.jds[-1] - spline.datapoints.jds[0])*2.0
 				xs = np.linspace(spline.datapoints.jds[0], spline.datapoints.jds[-1], npts)
 				ys = spline.eval(jds = xs)
-				plt.plot(xs, ys, "-", color=spline.plotcolour, zorder=+20, label=str(spline))
+				axes.plot(xs, ys, "-", color=spline.plotcolour, zorder=+20, label=str(spline))
 				# For the knots, we might not want to show them (by default we do show them) :
 				if getattr(spline, "showknots", True) == True:
-					#plt.errorbar(spline.getinttex(), spline.eval(jds = spline.getinttex()), 0.015*np.ones(len(spline.getinttex())), capsize=0, ecolor=spline.plotcolour, linestyle="none", marker="", elinewidth=1.5, zorder=40, barsabove=True)
-					
-					ax = plt.gca()
-					knotxs = spline.getinttex()
-					knotys = spline.eval(knotxs)
-					for (knotx, knoty) in zip(knotxs, knotys):
-						l = matplotlib.lines.Line2D([knotx,knotx],[knoty-knotsize,knoty+knotsize], zorder=30, linewidth=1.5, color=spline.plotcolour)                                    
- 						ax.add_line(l)  
+					if ax != None:
+						ax.errorbar(spline.getinttex(), spline.eval(jds = spline.getinttex()), 0.015*np.ones(len(spline.getinttex())), capsize=0, ecolor=spline.plotcolour, linestyle="none", marker="", elinewidth=1.5, zorder=40, barsabove=True)
```

Comment:
This line was commented out in the previous, and it seems that it was replaced by the loop below it. I guess that the reason for this is the zorder. 
Now it seems that you plot these "knot ticks" twice. Could you investigate and comment out the "errorbar" line again?